### PR TITLE
Refactor - HTTPServer upgrade handler

### PR DESCRIPTION
**Pull request by nethraravindran**
Refactored the code around `KituraWebSocketUpgrader` to improve readability

### CODE REVIEW

File: Sources/KituraNet/HTTP/HTTPServer.swift [hunk 1]
```diff
@@ -68,6 +68,8 @@ public class HTTPServer : Server {
     /// The event loop group on which the HTTP handler runs
     let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
 
+    private var channelHandlerCtx: ChannelHandlerContext?
```
Comment (pushkarnk) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: Considering SwiftNIO conventions, simply calling this `ctx` may make sense?

**Comment by pushkarnk** [comment]
@nethraravindran You may also want to run tests from the WebSocket NIO port locally, before we merge this.

**Comment by nethraravindran** [comment]
Few WebSocket tests failed with https://github.com/IBM-Swift/Kitura-NIO/pull/86 
Incorrect HTTPVersion in `sendResponse(channel: handler: status: withBody)` were the reasons for those failures. Defaulting to 1 fixes the issue.

**Comment by nethraravindran** [comment]
@pushkarnk Thanks! I have run WebSocket tests and all the tests are successful with this change https://github.com/IBM-Swift/Kitura-NIO/pull/88/commits/97772c6148e9494328fbb42c4ef2d5377345168d

**Comment by pushkarnk** [comment]
Thanks!

**Pull request merged**